src/app/contexts/UserContext/cookies/index.js

```diff
@@ -6,12 +6,21 @@ export const getCookiePolicy = () => {
   return Cookie.get(POLICY_COOKIE) || '000';
 };
 
+export const personalisationEnabled = cookiePolicy =>
+  cookiePolicy && cookiePolicy.length === 3 && cookiePolicy[2] === '1';
+
 export const getPreferredVariant = service => {
   if (!service) return null;
   const VARIANT_COOKIE = `ckps_${service}`;
 
   return Cookie.get(VARIANT_COOKIE);
 };
 
-export const personalisationEnabled = cookiePolicy =>
-  cookiePolicy && cookiePolicy.length === 3 && cookiePolicy[2] === '1';
+export const setPreferredVariantCookie = (service, variant) => {
+  if (!service || !variant || !personalisationEnabled(getCookiePolicy())) {
+    return;
+  }
+
+  const variantCookieName = `ckps_${service}`;
```

Comment:
Are we supposed to be setting this to `ckps_${service}`? In the linked issue, both zhongwen and ukchina are set to `ckps_chinese`.

Reply:
Can check the current cookies set by the PAL

Reply:
PAL uses `ckps_chinese`. I've reworked things to get this working.